client/signers.go

```diff
@@ -63,6 +69,8 @@ type (
 	Token interface {
 		// SetAuthHeader sets the Authorization header to r.
 		SetAuthHeader(r *http.Request)
+		// SetWsAuthHeader sets the Authorization header to cfg.
```

Comment:
Ah we need to think a bit more about this as it would be a breaking change... We may need to introduce a new `WSToken` interface as well. This may also mean a `WSTokenSource` interface :( I'm starting to wonder if a better approach wouldn't be to leave all of this the way it is and instead change the generated code so that it configures the websocket using the result of  signing a dummy request (i.e. extract the auth header and use it to initialize the websocket config).